spark-cassandra-connector/src/main/scala/org/apache/spark/metrics/CassandraConnectorSource.scala

```diff
@@ -20,8 +20,25 @@ object CassandraConnectorSource extends Source {
 
   val readByteMeter = metricRegistry.meter("read-byte-meter")
   val readRowMeter = metricRegistry.meter("read-row-meter")
-  val readPageWaitTimer = metricRegistry.timer("read-page-wait-timer")
   val readTaskTimer = metricRegistry.timer("read-task-timer")
 
-  lazy val ensureInitialized = SparkEnv.get.metricsSystem.registerSource(this)
+  CassandraConnectorSource._instance = Some(this)
+  CassandraConnectorSource._env = SparkEnv.get
```

Comment:
This class is instantiated by `SparkEnv` so there can be only one active instance, just like `SparkEnv`. I will make the constructor of this class package private for Spark metrics package. This should be enough. However the other thing I can think about is to make sure that when then new instance is created, either there is no instance associated with `_instance` or the current env is different than `_env`.